BCS.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+
+def consistent(M, b):
+    """
+    Check if the binary system Mx = b (mod 2) is consistent
+    """
+    aq = list(range(M.shape[0]))
+    A = M.copy()
+    b1 = b.copy()
+    m, n = A.shape
+    row = 0
+
+    for col in range(n):
+        if row >= m:
+            break
+        # Find a pivot row in the set defined by aq.
+        pivot = None
+        for r in range(row, m):
+            if A[aq[r], col] == 1:
+                pivot = r
+                break
+        if pivot is None:
+            continue 
+        # Swap the pivot row into the current position.
+        if pivot != row:
+            A[[aq[row], aq[pivot]]] = A[[aq[pivot], aq[row]]]
+            b1[[aq[row], aq[pivot]]] = b1[[aq[pivot], aq[row]]]
+        # Eliminate the 1's below the pivot in this column.
+        for r in range(row + 1, m):
+            if A[aq[r], col] == 1:
+                A[aq[r]] = (A[aq[r]] + A[aq[row]]) % 2
+                b1[aq[r]] = (b1[aq[r]] + b1[aq[row]]) % 2
+        row += 1
+    for r in range(m):
+        if np.all(A[aq[r]] == 0) and b1[aq[r]] == 1:
+            return False
+    return True
+
+class NonlocalGame:
```

Comment:
This is defined in the `nonlocal_games/` module. Is there are reason you are defining it in this file?